authenticated-fetch.ts

```diff
@@ -0,0 +1,38 @@
+export default async function authenticatedFetch(
+	url: string,
+	{signal}: {signal?: AbortSignal} = {},
+): Promise<Response> {
+	const token = globalThis.localStorage?.getItem('token');
+
+	const response = await fetch(url, {
+		...(token
+			? {
+				headers: {
+					// eslint-disable-next-line @typescript-eslint/naming-convention
+					Authorization: `Bearer ${token}`,
+				},
+			}
+			: {}),
+		signal,
+	});
+
+	switch (response.status) {
+		case 401: {
+			throw new Error('Invalid token');
+		}
+
+		case 403:
+		case 429: {
+			// See https://developer.github.com/v3/#rate-limiting
+			if (response.headers.get('X-RateLimit-Remaining') === '0') {
+				throw new Error('Rate limit exceeded');
```

Comment:
Every fetch handles rate limiting now, rather than just the "fetch repo info" one